RSS Reader/RSS Reader/Services/File Service/RSSRFileService.m

```diff
@@ -0,0 +1,73 @@
+//
+//  RSSRFileManager.m
+//  RSS Reader
+//
+//  Created by Lina Loyko on 1/12/21.
+//
+
+#import "RSSRFileService.h"
+
+static NSString * const kStorageFileName = @"RSSChannelStorage";
+
+@interface RSSRFileService ()
+
+@property (nonatomic, retain) NSString *storageFilePath;
+
+@end
+
+@implementation RSSRFileService
+
+- (NSString *)storageFilePath {
+    if (!_storageFilePath) {
+        NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+        _storageFilePath = [[[paths objectAtIndex: 0] stringByAppendingPathComponent: kStorageFileName] retain];
+    }
+    return _storageFilePath;
+}
+
+- (void)dealloc {
+    [_storageFilePath release];
+    [super dealloc];
+}
+
+- (void)saveChannel:(RSSRChannel *)channel completion:(void (^)(NSError *error))completion {
+    NSError *error = nil;
+    NSDictionary *jsonObject = [channel convertToDictionary];
+    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:jsonObject
+                                                       options:kNilOptions
+                                                         error:&error];
+    if (error) {
+        completion(error);
```

Comment:
You will hit a crash if completion is nil

Reply:
Fixed